### PR TITLE
cling 0.4 (new formula)

### PR DESCRIPTION
**Pull request by AndrewSB**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [ x Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

I'm having some trouble getting this to build, this might be WIP

### CODE REVIEW

**Comment by JCount** [comment]
Why isn't this using the main repository at https://github.com/root-project/cling?

**Comment by AndrewSB** [comment]
thats a great question, I didn't realize it wasn't @vgvassilev's project. I'll change it soon :)

**Comment by vgvassilev** [comment]
Yeah, both repos are auto synced with the true main repo https://github.com/root-project/root :). Either one should be fine.

File: Formula/cling.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Documentation: https://docs.brew.sh/Formula-Cookbook.html
+#                http://www.rubydoc.info/github/Homebrew/brew/master/Formula
+# PLEASE REMOVE ALL GENERATED COMMENTS BEFORE SUBMITTING YOUR PULL REQUEST!
```
Comment (fxcoudert) [inline Formula/cling.rb]: I am sure you will understand that reading in your pull request the comment saying `+# PLEASE REMOVE ALL GENERATED COMMENTS BEFORE SUBMITTING YOUR PULL REQUEST!` does nothing to persuade us that the formula will follow Homebrew's rules for new formulas…

File: Formula/cling.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+class Cling < Formula
+  desc "The interactive C++ interpreter Cling"
+  homepage "https://rawgit.com/vgvassilev/cling/master/www/index.html"
```
Comment (fxcoudert) [inline Formula/cling.rb]: Use the official ROOT webpage https://root.cern.ch/cling

File: Formula/cling.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  url "https://github.com/vgvassilev/cling/archive/v0.4.tar.gz"
```
Comment (fxcoudert) [inline Formula/cling.rb]: Please install from official sources

File: Formula/cling.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  sha256 "67b181309c5fe712dcd00903d48fb6965c46d517847d0b7f506a1921ba2f56b6"
+
+  depends_on "cmake" => :build
+  depends_on "llvm" => :build
+
+  resource "cling" do
+    url "https://github.com/vgvassilev/cling/archive/v0.4.tar.gz"
```
Comment (fxcoudert) [inline Formula/cling.rb]: Please install from official sources

File: Formula/cling.rb [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    sha256 "67b181309c5fe712dcd00903d48fb6965c46d517847d0b7f506a1921ba2f56b6"
+  end
+
+  resource "clang" do
+    url "https://github.com/vgvassilev/clang.git", :using => :git, :tag => "cling-patches-r302975", :revision => "1d3611b352ada6e8fc8dd41631bd17495f37b258"
```
Comment (fxcoudert) [inline Formula/cling.rb]: Please install from official sources

File: Formula/cling.rb [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+    mktemp do
+      system "cmake", "-G", "Unix Makefiles", buildpath, *std_cmake_args
+      system "make"
+      system "make", "install"
+    end
+  end
+
+  test do
+    system "#{bin}/cling", "-v"
```
Comment (fxcoudert) [inline Formula/cling.rb]: We need a test that actually does something

**Review by fxcoudert — CHANGES_REQUESTED** [review]

**Comment by fxcoudert** [comment]
```
Error: cling-patches-r302975 tag should be 1d3611b352ada6e8fc8dd41631bd17495f37b258
but is actually 280204113c744b685279a4c458cbdfff95a6a829
```

```
cling:
  * C: 21: col 13: sha256 should be 64 characters
  * Dependency 'llvm' may be unnecessary as it is provided by macOS; try to build this formula without it.
  * Please remove default template comments
  * Please remove default template comments
```

**Pull request closed without merging**